config/v1/types_network.go

```diff
@@ -70,6 +74,16 @@ type NetworkSpec struct {
 	// installed.
 	// +kubebuilder:validation:Pattern=`^([0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])-([0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$`
 	ServiceNodePortRange string `json:"serviceNodePortRange,omitempty"`
+
+	// networkDiagnostics defines network diagnostics configuration.
+	//
+	// Cannot be specified when spec.disableNetworkDiagnostics is set to true in network.operator.openshift.io.
```

Comment:
This looks backwards.  The configuration in config.openshift.io should be authoritative.

Reply:
Changed it so networkDiagnostics takes precedence over disableNetworkDiagnostics if specified.